ndcube/ndcube.py

```diff
@@ -1035,37 +1035,43 @@ def __neg__(self):
         return self._new_instance(data=-self.data)
 
     def __add__(self, value):
+        kwargs = {}
         if isinstance(value, NDData) and value.wcs is None:
             if self.unit is not None and value.unit is not None:
-                    value_data = value.data * value.unit.to(self.unit)
+                    value_data = (value.data * value.unit).to_value(self.unit)
             elif self.unit is None:
                 value_data = value.data
             else:
                 raise TypeError("Cannot add unitless NDData to a unitful NDCube.")
 
-            # addition
-            new_data = self.data + value_data
             # combine the uncertainty
-            new_uncertainty = None
             if self.uncertainty is not None and value.uncertainty is not None:
                 new_uncertainty = self.uncertainty.propagate(
-                    np.add, value.uncertainty, correlation=0
+                    np.add, value.uncertainty, result_data = value.data, correlation=0
                 )
+                kwargs["uncertainty"] = new_uncertainty
             elif self.uncertainty is not None:
                 new_uncertainty = self.uncertainty
+                kwargs["uncertainty"] = new_uncertainty
             elif value.uncertainty is not None:
                 new_uncertainty = value.uncertainty
+            else:
+                new_uncertainty = None
 
             # combine mask
             self_ma = np.ma.MaskedArray(self.data, mask=self.mask)
             value_ma = np.ma.MaskedArray(value_data, mask=value.mask)
+
+            # addition
             result_ma = self_ma + value_ma
-            new_mask = result_ma.mask
+
+            # extract new mask and new data
+            kwargs["mask"] = result_ma.mask
+            kwargs["data"] = result_ma.data
```

Comment:
As mentioned in above comment, I think it makes sense to do this before the uncertainty propagation so you can use the `kwargs["data"]` value in that propagation.